NewHaven/src/VGMap.cpp

```diff
@@ -349,14 +347,14 @@ Circle & Circle::operator=(const Circle &circle){
     if (this == &circle)
         return *this;
     else {
-        *row = *circle.row;
-        *column = *circle.column;
-        *vCost = *circle.vCost;
+        row = new int(*circle.row);
+        column = new int(*circle.column);
+        vCost = new int(*circle.vCost);
```

Comment:
Good catch, you can also use list initialization in copy constructors if you want.